tools/analysis_tools/analyze_results.py

```diff
@@ -110,33 +121,49 @@ def _save_image_gts_results(self,
 
         for performance_info in performances:
             index, performance = performance_info
-            data_info = dataset.prepare_train_img(index)
+            data_info = dataset[index]
+            data_info['gt_instances'] = data_info['instances']
 
             # calc save file path
-            filename = data_info['filename']
-            if data_info['img_prefix'] is not None:
-                filename = osp.join(data_info['img_prefix'], filename)
-            else:
-                filename = data_info['filename']
+            filename = data_info['img_path']
             fname, name = osp.splitext(osp.basename(filename))
             save_filename = fname + '_' + str(round(performance, 3)) + name
             out_file = osp.join(out_dir, save_filename)
-            imshow_gt_det_bboxes(
+
+            if task == 'det':
+                gt_instances = InstanceData()
+                gt_instances.bboxes = data_info['gt_bboxes']
+                gt_instances.labels = data_info['gt_bboxes_labels']
+                gt_samples = DetDataSample()
+                gt_samples.gt_instances = gt_instances
+
+                pred_instances = InstanceData()
+                pred_instances.bboxes = results[index]['pred_instances'][
+                    'bboxes']
+                pred_instances.labels = results[index]['pred_instances'][
+                    'labels']
+                pred_instances.scores = results[index]['pred_instances'][
+                    'scores']
+                pred_samples = DetDataSample()
+                pred_samples.pred_instances = pred_instances
+            elif task == 'seg':
+                gt_samples = DetDataSample()
+                gt_sem_seg_data = dict(
+                    sem_seg=to_tensor(data_info['gt_seg_map']))
+                gt_samples.gt_sem_seg = PixelData(**gt_sem_seg_data)
+
+                pred_samples = DetDataSample()
+                pred_samples.pred_panoptic_seg = PixelData(
+                    sem_seg=results[index]['pred_panoptic_seg']['sem_seg'])
+
+            self.visualizer.add_datasample(
+                'image',
                 data_info['img'],
-                data_info,
-                results[index],
-                dataset.CLASSES,
-                gt_bbox_color=dataset.PALETTE,
-                gt_text_color=(200, 200, 200),
-                gt_mask_color=dataset.PALETTE,
-                det_bbox_color=dataset.PALETTE,
-                det_text_color=(200, 200, 200),
-                det_mask_color=dataset.PALETTE,
+                gt_samples,
```

Comment:
visualizer will be update in  https://github.com/open-mmlab/mmdetection/pull/8620